src/System.IO.Packaging/tests/Tests.cs

```diff
@@ -3639,6 +3639,109 @@ public void SetEmptyPropertyToNull()
+
+                        using (var partStream = part.GetStream())
+                        using (var reader = new StreamReader(partStream, Encoding.ASCII))
+                        {
+                            Assert.Equal(fileName.Length, partStream.Length);
+                            Assert.Equal(fileName, reader.ReadToEnd());
+                        }
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        [OuterLoop]
+        public void VeryLargePart()
+        {
+            // FileAccess.Write is important, this tells ZipPackage to open the underlying ZipArchive in 
+            // ZipArchiveMode.Create mode as opposed to ZipArchiveMode.Update
+            // When ZipArchive is opened in Create it will write entries directly to the zip stream
+            // When ZipArchive is opened in Update it will write uncompressed data to memory until 
+            // the archive is closed.
+            using (var stream = new MemoryStream())
+            {
+                Uri partUri = PackUriHelper.CreatePartUri(new Uri("test.bin", UriKind.Relative));
+
+                // should compress *very well*
+                byte[] buffer =  new byte[1024 * 1024];
+                for(int i = 0; i < buffer.Length; i++)
+                {
+                    buffer[i] = (byte)(i % 2);
+                }
+                
+                const long sizeInMb = 6 * 1024; // 6GB
+                long totalLength = sizeInMb * buffer.Length;
+
+                using (Package package = Package.Open(stream, FileMode.Create, FileAccess.Write))  
+                {
+                    PackagePart part = package.CreatePart(partUri,
+                                                          System.Net.Mime.MediaTypeNames.Application.Octet,
+                                                          CompressionOption.Fast);
+
+
+                    using (Stream partStream = part.GetStream())
+                    {
+                        for (long i = 0; i < sizeInMb; i++)
+                        {
+                            partStream.Write(buffer, 0, buffer.Length);
+                        }
+                    }
+                }
+
+                // reopen for read and make sure we can get the part length & data matches
+                stream.Seek(0, SeekOrigin.Begin);
+                using (Package readPackage = Package.Open(stream))
+                {
+                    var part = readPackage.GetPart(partUri);
+
+                    using (Stream partStream = part.GetStream())
+                    {
+                        Assert.Equal(totalLength, partStream.Length);
+
+                        byte[] readBuffer = new byte[buffer.Length];
+                        for(long i = 0; i < sizeInMb; i++)
+                        {
+                            int actualRead = partStream.Read(readBuffer, 0, readBuffer.Length);
+
+                            Assert.Equal(actualRead, readBuffer.Length);
```

Comment:
Nit: swap the arguments so that it matches expected,actual

Reply:
Is it actually guaranteed to read the full amount requested?  That's not generally guaranteed for streams, but maybe it is in this specific case?

Reply:
The implementation of DeflateStream ensures this behavior, it'll only return less if it reaches the end of the compressed data or reaches the end of backing stream: 
https://github.com/dotnet/corefx/blob/76e08b605f237bffb74283d83241eaadc4e7ce93/src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs#L262-L297

Reply:
Ok, thanks.

Reply:
Your comment is good for production code.  In this case I think I'd want to know if we changed this behavior, since its reasonably observable and consistent between desktop and core today.